app/gui/src/controller/searcher/component/group.rs

```diff
@@ -118,6 +118,20 @@ impl Group {
         })
     }
 
+    /// Create a new [`Group`] retaining only those [`Component`]s in the [`initial_entries_order`]
+    /// vector for which `f` returns [`true`]. The [`entries`] are set to a clone of the vector,
+    /// and [`matched_items`] are set to the length of the vector.
+    pub fn with_entries_in_initial_order_and_filtered<F>(self, f: F) -> Self
```

Comment:
1. Why function named `with_entries_in_initial_order_and_filtered` creates a new group (docs say so)? If it creates a new group, it should start with "new". (it consumes self, so it does not create a new group, so the docs are wrong. It consumes and modifies self).
2. "The [`entries`] are set to a clone of the vector" – implementation detail that this is vector clone. After we change the implementation, we would need to update the docs.
3. "[`matched_items`] are set to the length of the vector" - what does it mean?
4. There is no high-level explanation of the purpose of this function. I feel that the docs explain a pseudo-algorithm, but not the purpose of the function.
5. What is "initial order"?